diff --git a/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/Makefile b/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/Makefile
index d0970cb..656af12 100644
--- a/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/Makefile
+++ b/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/Makefile
@@ -1,6 +1,6 @@
 EXTRA_CFLAGS += $(USER_EXTRA_CFLAGS)
 EXTRA_CFLAGS += -Wno-implicit-fallthrough
-#EXTRA_CFLAGS += -Wno-unused-variable
+EXTRA_CFLAGS += -Wno-unused-variable
 
 RWNX_VERS_NUM := 6.4.3.0
 
diff --git a/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_mod_params.c b/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_mod_params.c
index f78ca5f..3e7891f 100644
--- a/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_mod_params.c
+++ b/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_mod_params.c
@@ -936,7 +936,7 @@ static void rwnx_set_vht_capa(struct rwnx_hw *rwnx_hw, struct wiphy *wiphy)
         nss = 1;
     }
 
-	band_2GHz->vht_cap.vht_supported = true;
+    band_2GHz->vht_cap.vht_supported = true;
 		if (rwnx_hw->mod_params->sgi80)
 			band_2GHz->vht_cap.cap |= IEEE80211_VHT_CAP_SHORT_GI_80;
 		if (rwnx_hw->mod_params->stbc_on)
