Index: aic8800/src/SDIO/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c
===================================================================
--- aic8800.orig/src/SDIO/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c
+++ aic8800/src/SDIO/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c
@@ -3763,6 +3763,9 @@ static int rwnx_cfg80211_set_tx_power(st
 static int rwnx_cfg80211_get_tx_power(struct wiphy *wiphy,
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 8, 0)
  struct wireless_dev *wdev,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION (6, 14, 0)
+ unsigned int link_id,
+#endif
 #endif
 	int *mbm)
 {
Index: aic8800/src/PCIE/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c
===================================================================
--- aic8800.orig/src/PCIE/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c
+++ aic8800/src/PCIE/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c
@@ -3845,6 +3845,9 @@ static int rwnx_cfg80211_set_tx_power(st
 static int rwnx_cfg80211_get_tx_power(struct wiphy *wiphy,
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 8, 0)
  struct wireless_dev *wdev,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION (6, 14, 0)
+ unsigned int link_id,
+#endif
 #endif
 	int *mbm)
 {
Index: aic8800/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c
===================================================================
--- aic8800.orig/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c
+++ aic8800/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c
@@ -4513,6 +4513,9 @@ int radio_idx,
 static int rwnx_cfg80211_get_tx_power(struct wiphy *wiphy,
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 8, 0)
  struct wireless_dev *wdev,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION (6, 14, 0)
+ unsigned int link_id,
+#endif
 #endif
 	int *mbm)
 {
