diff --git a/src/USB/driver_fw/drivers/aic_btusb/aic_btusb.c b/src/USB/driver_fw/drivers/aic_btusb/aic_btusb.c
index 1fad30f..a69ed8a 100644
--- a/src/USB/driver_fw/drivers/aic_btusb/aic_btusb.c
+++ b/src/USB/driver_fw/drivers/aic_btusb/aic_btusb.c
@@ -1409,6 +1409,7 @@ static long btchr_ioctl(struct file *file_p,unsigned int cmd, unsigned long arg)
             return 1;
         case DWFW_CMPLT:
             AICBT_INFO(" btchr_ioctl DWFW_CMPLT");
+            __attribute__((__fallthrough__));
 #if 1
 	case SET_ISO_CFG:
             AICBT_INFO("btchr_ioctl SET_ISO_CFG");
@@ -1418,6 +1419,7 @@ static long btchr_ioctl(struct file *file_p,unsigned int cmd, unsigned long arg)
 		//hdev->voice_setting = *(uint16_t*)arg;
 		AICBT_INFO(" voice settings = %d", hdev->voice_setting);
 		//return 1;
+        __attribute__((__fallthrough__));
 #endif
         case GET_USB_INFO:
 			//ret = download_patch(fw_info,1);
@@ -1446,14 +1448,6 @@ static long btchr_ioctl(struct file *file_p,unsigned int cmd, unsigned long arg)
 
 }
 
-#ifdef CONFIG_PLATFORM_UBUNTU//AIDEN
-typedef u32		compat_uptr_t;
-static inline void __user *compat_ptr(compat_uptr_t uptr)
-{
-	return (void __user *)(unsigned long)uptr;
-}
-#endif
-
 #ifdef CONFIG_COMPAT
 static long compat_btchr_ioctl (struct file *filp, unsigned int cmd, unsigned long arg)
 {
